public/javascript/posts/BookAppointment.js

```diff
@@ -281,19 +281,58 @@ var app = angular.module('BookAppointmentApp');
           end: $scope.selected_event.end,
           className: ['openSesame'],
           stick: true,
-          backgroundColor: backgroundColor,
+          backgroundColor: backgroundColor,
           event_type: event_type
-        });
+        });
+      };
+
+      $scope.getDataToSend = function(event_hash){
+        var hash = {};
+        hash['doctorId'] = event_hash.doctor_id;
+        hash['doctorlocationId'] = event_hash.location_id;
+        hash['appointmentStartTime'] = event_hash.start;
+        hash['appointmentEndTime'] = event_hash.end;
+        hash['appointmentTitle'] = event_hash.subject;
+        hash['cancelOverlapped'] = event_hash.cancel_overlapped_event;
+        hash['isAllDayEvent'] = event_hash.is_all_day_event;
+        hash['cretaedDate'] = event_hash.cretaed_date;
+        hash['createdby'] = event_hash.created_by;
+
+        if(event_hash.event_type == 'booking'){
+          hash['patname'] = event_hash.patient_name;
+          hash['mobileno'] = event_hash.mobile_number;
+          hash['prepayAmount'] = event_hash.prepay_amount;
+          hash['prepayBy'] = event_hash.prepay_by;
+          hash['email'] = event_hash.email;
+          hash['patienttype'] = event_hash.patient_type;
+          hash['patientId'] = "114";
+        }else if(event_hash.event_type == 'blocked'){
+          hash['appointmentType'] = event_hash.appointment_type;
+        }
+        return hash;
+      };
+
+      $scope.bookAppointment = function(url_to_post, event_hash){
+        var data = $scope.getDataToSend(event_hash);
+        $http.post(url_to_post, data).success(function(response){
+          if(response.IsSuccess){
+            $scope.addEvent(response.event_id);
+            $scope.appointmentBooked();
+          }else{
+            $scope.appointmentNotBooked();
+          }
+        });
       };
 
       modalInstance.result.then(function (selectedItem) {
         var event_hash = {};
+        var url_to_post = 'http://connect.s.miraihealth.com/CalendarService/CalendarService.svc/AddAppointment';

```

Comment:
instead of passing the `url_to_post` declare and use it in `bookAppointment` function directly